src/NUnitFramework/framework/Constraints/CollectionEquivalentConstraintResult.cs

```diff
@@ -0,0 +1,113 @@
+    /// </summary>
+    /// <remarks>
+    /// Assumes that a <see cref="CollectionTally"/> has been created and used to compare the expected
+    /// equivalent collections.
+    /// </remarks>
+    public class CollectionEquivalentConstraintResult : ConstraintResult
+    {
+        /// <summary>
+        /// Preprocessed <see cref="CollectionTally"/> of the collections to compare for equivalence.
+        /// </summary>
+        private CollectionTally _tally;
+
+        /// <summary>
+        /// Maximum amount of elements to write to the <see cref="MessageWriter"/> if there are
+        /// extra/missing elements from the collection.
+        /// </summary>
+        private const int MaxDifferingElemsToWrite = 10;
+        
+        /// <summary>
+        /// Construct a <see cref="CollectionEquivalentConstraintResult"/> with a <see cref="CollectionTally"/>
+        /// already applied to the actual collection.
+        /// </summary>
+        /// <param name="Constraint">
+        /// Source <see cref="CollectionEquivalentConstraint"/>.
+        /// </param>
+        /// <param name="Tally">
+        /// Preprocessed <see cref="CollectionTally"/> for the collections to compare.
+        /// </param>
+        /// <param name="actual">
+        /// Collection to compare to the expected one.
+        /// </param>
+        /// <param name="isSuccess">
+        /// Whether or not the <see cref="CollectionEquivalentConstraint"/> succeeded.
+        /// </param>
+        /// <remarks>
+        /// The already processed <see cref="CollectionTally"/> is provided to reduce
+        /// unnecessary iterations over the collections to determine equivalence.
+        /// </remarks>
+        public CollectionEquivalentConstraintResult(
+            CollectionEquivalentConstraint Constraint,
+            CollectionTally Tally,
+            object actual,
+            bool isSuccess)
+            : base(Constraint, actual, isSuccess)
+        {
+            if (Tally == null)
+            {
+                throw new System.ArgumentNullException("Tally was null.");
+            }
+
+            _tally = Tally;
+        }
+
+        /// <summary>
+        /// Write the custom failure message for this object's 
+        /// <see cref="CollectionEquivalentConstraint"/>.
+        /// </summary>
+        /// <param name="writer"></param>
+        public override void WriteMessageTo(MessageWriter writer)
+        {
+            //Write the expected/actual message first.
+            base.WriteMessageTo(writer);
+            
+            if (_tally.MissingItems.Count > 0)
+            {
+                int missingItemsCount = _tally.MissingItems.Count;
+
+                string missingStr = $"Missing ({missingItemsCount}): ";
+                missingStr += MsgUtils.FormatCollection(_tally.MissingItems, 0, MaxDifferingElemsToWrite);
+
+                writer.WriteMessageLine(missingStr);
+            }
+
+            if (_tally.ExtraItems.Count > 0)
```

Comment:
@OmicronPersei - good job checking. We're not on the same page! 😄 

I agree what you currently have is good. The lines I **wouldn't** want to see are:

> Missing(0): < empty >

or

> Extra(0): < empty >

From looking at the code, this is what you've already done? I agree `Expected: <empty>` is the right thing to show.

My suggestion was that, I don't believe there are any tests that `Extra(0): < empty >` is _not_ shown when there are no extra items? I think all your tests have both Extra AND missing items? I may have just missed the test however, in which case, apologies!